### PR TITLE
[core] Use geometry.hpp's point type

### PR DESCRIPTION
**Pull request by jfirebaugh**

### CODE REVIEW

**Comment by jfirebaugh** [comment]
cc @kkaefer @springmeyer

File: platform/osx/osx.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1018,6 +1022,10 @@
 				HEADER_SEARCH_PATHS = ../../include;
 				INFOPLIST_FILE = test/Info.plist;
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/../Frameworks @loader_path/../Frameworks";
+				OTHER_CPLUSPLUSFLAGS = (
+					"$(OTHER_CFLAGS)",
+					"$(geometry_cflags)",
+				);
```
Comment (springmeyer) [inline platform/osx/osx.xcodeproj/project.pbxproj]: @jfirebaugh how come we need this here in addition to https://github.com/mapbox/mapbox-gl-native/blob/geometry.hpp/platform/osx/osx.xcodeproj/project.pbxproj#L985?

**Comment by springmeyer** [comment]
overall 💯

File: platform/osx/osx.xcodeproj/project.pbxproj (hunk 1 above)
Comment (jfirebaugh) [inline platform/osx/osx.xcodeproj/project.pbxproj]: There are two targets which wind up including mapbox/geometry.hpp (via map.hpp): this one is the osx `test` target, and the other one is the `dynamic` target that builds a framework.

**Pull request merged**